### ISSUE TITLE
govcms_tags module contains an undeclared dependency

### ISSUE THREAD

**Issue by invisigoth** [govCMS/GovCMS7#96]
govcms_tags module requires zen base theme to function properly and may throw a WSOD on tag pages if zen is not enabled.

This is caused by a hard-coded panel layout in line 37 of govcms_tags.pages_default.inc:
$display->layout = 'zen_no_wrapper';

It was noticed while building blog.data.gov.au.

**Comment by fiasco** [comment]
I don't believe Drupal has a way to making a theme a dependancy of a module. Options:
- Provide a layout dynamically selected by theme?
- Enforce the use of a zen base in govCMS?
- Can we remove the dependancy from govCMS?

**Comment by invisigoth** [comment]
Hi Josh,

Since we have little background of this feature and it has not been documented anywhere that Zen has to be enabled for GovCMS to be fully functional, the concern is for those trying to use GovCMS with their custom theme may run into issue and it may be difficult to debug. See commit 9449356.

Perhaps it's simply in the documentation alerting theme developers that they need to enable Zen.

Thank you!

**Comment by pandaskii** [comment]
We already have few sites using bootstrap. "Enable Zen" may bring conflicts

We can either create a layout such as "govcms_no_wrapper" or re-design govcms_tags module (remove Zen panel layout dependancy)

**Comment by fiasco** [comment]
A more broader question is, why are we building themes that are not built on the govCMS base theme? It seems to be creating issues that govCMS was not designed to handle. While govCMS is built on Drupal which provides the flexibility to to use whatever theme you like, consolidating on a single theme base reduces what the govCMS community need to support and maintain and increases govCMS's stability. This issue is case and point of that.

That being said, my preference would be for bootstrap over zen. Its more widely supported and understood by those who don't use Drupal. So if we can bring in bootstrap as a second base theme option and then publish a constraint to use Zen or Bootstrap in govCMS then I think it gives us an easy direction on this issue which would be to convert govcms_tags to use a layout agnostic to both themes.

**Comment by invisigoth** [comment]
@fiasco Absolutely agree. Both blog.data.gov.au and australia.gov.au already use an in-house light-weight Bootstrap theme and can be readily adopted as a base theme.

However in the interim it could be beneficial to let the community know that there is a "hidden" panel layout dependency.  It could save other developers hours in case they run into the similar issue.

**Comment by pandaskii** [comment]
cannot agree more with @fiasco on bootstrap :)

"a constraint to use Zen or Bootstrap in govCMS "
- like +1

Fontawesome? even better

**Comment by fiasco** [comment]
Cool, I'd be happy if you guys proceeded with a fix to govcms_tags that worked for both zen and bootstrap. Would also be good to start a wiki page talking about our decisions around choice of theme use in govCMS.

**Comment by pandaskii** [comment]
Link this with #215
